mittab/libs/data_import/export_xls_files.py

```diff
@@ -0,0 +1,220 @@
+
+    headers = ['team name','# wins', 'total speaker points', 'total ranks', 'single-adjusted speaker points',
+               'single-adjusted ranks', 'double-adjusted speaker points', 'double-adjusted ranks',
+               'opposition strength']
+    for i in xrange(len(headers)):
+        sheet.write(1, i, headers[i])
+
+    teams = Team.objects.all()
+    for i in xrange(len(teams)):
+        team, row = teams[i], i + 2
+        sheet.write(row, 0, team.name)
+        sheet.write(row, 1, tab_logic.tot_wins(team))
+        sheet.write(row, 2, tab_logic.tot_speaks(team))
+        sheet.write(row, 3, tab_logic.tot_ranks(team))
+        sheet.write(row, 4, tab_logic.single_adjusted_speaks(team))
+        sheet.write(row, 5, tab_logic.single_adjusted_ranks(team))
+        sheet.write(row, 6, tab_logic.double_adjusted_speaks(team))
+        sheet.write(row, 7, tab_logic.double_adjusted_ranks(team))
+        sheet.write(row, 8, tab_logic.opp_strength(team))
+
+    # construct response
+    response = HttpResponse(content_type='application/vnd.ms-excel')
+    response['Content-Disposition'] = 'attachment; filename=mittab-team-stats.xls'
+    book.save(response)
+    return response
+
+
+@permission_required('tab.tab_settings.can_change', login_url="/403/")
+def export_debater_stats(request):
+    """Exports data as XLS for each speaker: total speaker points, total ranks, single adjusted speaks, single adjusted
+    ranks, double adjusted speaks, double adjusted ranks, team performance, opposition strength. Automatically averages
+    for iron-men. """
+    book = Workbook('utf-8')
+    sheet = book.add_sheet('Team stats')
+
+    sheet.write(0, 0, 'If you are calculating the awards off these statistics, please pay attention to how you sort. '
+                      'In general, everything except ranks should be sorted from large to small.')
+
+    headers = ['debater name', 'total speaker points', 'total ranks', 'single-adjusted speaker points',
+               'single-adjusted ranks', 'double-adjusted speaker points', 'double-adjusted ranks',
+               'team performance (debater\'s team win#)', 'opposition strength']
+    for i in xrange(len(headers)):
+        sheet.write(1, i, headers[i])
+
+    debaters = Debater.objects.all()
+    for i in xrange(len(debaters)):
+        debater, row = debaters[i], i + 2
+        sheet.write(row, 0, debater.name)
+        sheet.write(row, 1, tab_logic.tot_speaks_deb(debater, average_ironmen=True))
+        sheet.write(row, 2, tab_logic.tot_ranks_deb(debater, True))
+        sheet.write(row, 3, tab_logic.single_adjusted_speaks_deb(debater))
+        sheet.write(row, 4, tab_logic.single_adjusted_ranks_deb(debater))
+        sheet.write(row, 5, tab_logic.double_adjusted_speaks_deb(debater))
+        sheet.write(row, 6, tab_logic.double_adjusted_ranks_deb(debater))
+        sheet.write(row, 7, tab_logic.tot_wins(Team.objects.filter(debaters__name=debater.name).first()))
```

Comment:
Should use debater id rather than name for this search. People can have the same name but they can't get the same id.

You can also use [get()](https://docs.djangoproject.com/en/2.0/topics/db/queries/#retrieving-a-single-object-with-get) instead of filter() here, so you don't have to call first()